tests/test_predictor.py

```diff
@@ -17,8 +18,9 @@ def test_boston_dataset():
     X_train_binned = mapper.fit_transform(X_train)
     X_test_binned = mapper.transform(X_test)
 
-    gradients = y_train.astype(np.float32)
-    hessians = np.ones(1, dtype=np.float32)
+    loss = LeastSquares()
+    gradients = loss.get_gradients(y_train, 0)
+    hessians = loss.get_hessians(y_train, 0)
```

Comment:
This is not related to the predictor module. Please create a new test file named `test_losses.py` dedicated to unit test for the loss classes and then another test file named `test_gradient_boosting.py` that runs high level test for the estimator public API.

Reply:
I wasn't trying to test the loss here, I just had to change the hardcoded gradients / hessians computation for the test to pass.

This test implicitly used a least squares loss, and as I changed the way the leaf values are computed, I also had to change the gradient / hessians computation for the test to pass. I can change it to 

```
    gradients = -2 * y_train
    hessians = np.full(shape=1, fill_value=2)
```
but that's still implicitely using least squares.

Of course I will subsequently add `test_losses.py` and `test_gradient_boosting`.

Reply:
Alright.